### PR TITLE
Schedule Keyword Handler Dispatch

### PR DESCRIPTION
**Pull request by wito**
This feature replaces the current heavily `else if`-based dispatch method used in `Schedule::handleKeyword` with an `std::unordered_map<std::string, std::function>`-based lookup, as part of move towards a single, clear path through the handler for the purposes of error handling.

### CODE REVIEW

**Comment by wito** [comment]
jenkins build this please

**Comment by blattms** [comment]
Can we please use git rebase instead merging master into a PR-branch?

**Comment by wito** [comment]
@blattms Unfortunately, due to an error I made when forking this branch (forking it from a months-old master), and the deep cutting nature of the changes, rebasing of this feature is an intractable problem.

**Comment by wito** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
Here: https://github.com/OPM/opm-common/pull/1947 is a 95% rebased version - cool if you complete the rebasing here so that we avoid the merges in the current branch

**Comment by wito** [comment]
jenkins build this please

**Comment by joakim-hove** [comment]
jenkins build this please

File: src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+#include "Well/injection.hpp"
+
+namespace Opm {
+
+namespace {
+
+}
+
+    void Schedule::handleMESSAGES(const HandlerContext& handlerContext, const ParseContext&, ErrorGuard&) {
```
Comment (joakim-hove) [inline src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp:86]: Most of the `handleXXXX`functions are still in `Schedule.cpp`- is that on purpose - or just work in progress?

Comment (wito) [reply src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp:86]: The diff on this is already substantial; I'll be making a second PR that's _just_ moving all the handlers from one file to the other.

Comment (joakim-hove) [reply src/opm/parser/eclipse/EclipseState/Schedule/KeywordHandlers.cpp:86]: Fine - then I'll merge in current state.

**Pull request merged**